spec/components/s3_inventory_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+require "s3_inventory"
+require "file_store/s3_store"
+
+describe "S3Inventory" do
+  let(:store) { FileStore::S3Store.new }
+  let(:inventory) { store.inventory }
+  let(:csv_filename) { File.new("#{Rails.root}/spec/fixtures/csv/s3_inventory.csv") }
+
+  before do
+    SiteSetting.enable_s3_uploads = true
+    SiteSetting.s3_access_key_id = "abc"
+    SiteSetting.s3_secret_access_key = "def"
+    SiteSetting.s3_inventory_bucket = "ghi"
+    SiteSetting.enable_s3_inventory = true
+
+    s3_helper = store.inventory.instance_variable_get(:@s3_helper)
+    client = Aws::S3::Client.new(stub_responses: true)
+    client.stub_responses(:list_objects, {
+      contents: [
+        {
+          etag: "\"70ee1738b6b21e2c8a43f3a5ab0eee71\"", 
+          key: "example1.csv.gz", 
+          last_modified: Time.parse("2014-11-21T19:40:05.000Z"), 
+          owner: {
+            display_name: "myname", 
+            id: "12345example25102679df27bb0ae12b3f85be6f290b936c4393484be31bebcc", 
+          }, 
+          size: 11, 
+          storage_class: "STANDARD",
+        }, 
+        {
+          etag: "\"9c8af9a76df052144598c115ef33e511\"", 
+          key: "example2.csv.gz", 
+          last_modified: Time.parse("2013-11-15T01:10:49.000Z"), 
+          owner: {
+            display_name: "myname", 
+            id: "12345example25102679df27bb0ae12b3f85be6f290b936c4393484be31bebcc", 
+          }, 
+          size: 713193, 
+          storage_class: "STANDARD", 
+        }
+      ],
+      next_marker: "eyJNYXJrZXIiOiBudWxsLCAiYm90b190cnVuY2F0ZV9hbW91bnQiOiAyfQ=="
+    })
+    s3_helper.stubs(:s3_client).returns(client)
+    Discourse.stubs(:store).returns(store)
```

Comment:
Hmm why do we need to stub this?

Reply:
Because `Discourse.store` method currently returning new `s3_store` object on every calls. Yes, the existing code needs to be changed. I'll improve this while s3_store and s3_helper refactoring.